### DISCUSSION TITLE
How to check if request is authenticated or not?

### DISCUSSION THREAD

**Discussion by tanujrawat95** [saleor/saleor#12278] [Q&A]
Hi!
I am trying to add one more app in saleor api(django) to download some files.
I have mentioned the app name in django settings, defined the url and the view for the app.
It is working fine but the problem I am facing now is how to check if the incoming request is authenticated or not.
I want only authenticated user to download those files.
Is there any decorator or function in saleor through which i can authenticate the incoming request and if the request is unauthenticated, I will deny the request.
We will have the jwt token in request which can be checked in saleor authentication workflow and oidc plugin workflow as well.

Is it doable without doing much of code modification?

**Comment by timuric** [comment]
Hi @tanujrawat95 our recommended approach using Saleor via GraphQL API, and we will be deprecating plugins very soon

**Reply by tanujrawat95** [reply]
Oh alright, thanks for the info. Will plan according to that
